fineract-e2e-tests-core/src/test/java/org/apache/fineract/test/helper/ErrorMessageHelper.java

```diff
@@ -348,9 +348,18 @@ public static String wrongDataInDelinquentLastRepaymentDate(String actual, Strin
     }
 
     public static String wrongLoanStatus(Integer actual, Integer expected) {
+        return wrongLoanStatus(null, actual, expected);
+    }
+
+    public static String wrongLoanStatus(String resourceId, Integer actual, Integer expected) {
         String actualToStr = actual.toString();
         String expectedToStr = expected.toString();
-        return String.format("Wrong Loan status ID. Actual ID is: %s - But expected ID is: %s", actualToStr, expectedToStr);
+        String prefx = "Wrong Loan status ID";
+        String postfx = ". Actual ID is: %s - But expected ID is: %s";
+        if (resourceId != null) {
+            return String.format(prefx + " of resource %s" + postfx, resourceId, actualToStr, expectedToStr);
+        }
+        return String.format(prefx + postfx, actualToStr, expectedToStr);
```

Comment:
I'm not a big fan of this string joining. Why didn't we just have the 2 messages inline?

Reply:
Agree, me neither.